File: src/main/java/com/bot/db/mappers/GuildMembershipMapper.java
package com.bot.db.mappers;

import com.bot.models.InternalGuildMembership;

import java.sql.ResultSet;
import java.sql.SQLException;

public class GuildMembershipMapper {
    public static InternalGuildMembership mapGuildMembership(ResultSet set) throws SQLException {
        return new InternalGuildMembership(set.getString("gm.user_id"),
                set.getString("gm.guild"),
                set.getBoolean("gm.can_use_bot"));
    }
}

File: src/main/java/com/bot/db/mappers/GuildMembershipMapper.kt
package com.bot.db.mappers

import com.bot.models.InternalGuildMembership
import java.sql.ResultSet
import java.sql.SQLException

object GuildMembershipMapper {
    @JvmStatic
    @Throws(SQLException::class)
    fun mapGuildMembership(set: ResultSet): InternalGuildMembership {
        return InternalGuildMembership(set.getString("gm.user_id"),
                set.getString("gm.guild"),
                set.getBoolean("gm.can_use_bot"))
    }
}